internal/sdkprovider/service/kafkaschema/kafka_schema.go

```diff
@@ -20,8 +20,9 @@ import (
 	"github.com/aiven/terraform-provider-aiven/internal/schemautil/userconfig"
 )
 
-// newlineRegExp is a regular expression that matches a newline.
-var newlineRegExp = regexp.MustCompile(`\r?\n`)
+// whitespaceNewLineCarriageReturnRegExp is a regular expression to match any whitespace, new line, or carriage return
+// characters in a string. This is used to normalize Protobuf strings to compare them for logical equivalence.
+var whitespaceNewLineCarriageReturnRegExp = regexp.MustCompile(`[\s\r\n]+`)
```

Comment:
Isn't it just `\s+`?

Reply:
Nope

Reply:
Changed it to `\s`, it also works because of `ReplaceAllString` call.

Reply:
On 100 line protobuf file, which is not a big one even, with spaces instead tabs it will produce 4 * per level including newlines. It might make hundreds and thousands loops because of this plus sign. Please add it.

Reply:
It works faster with this pattern, please see this benchmark, Protobuf used for benchmarking is from [here](https://raw.githubusercontent.com/SteamDatabase/Protobufs/master/steam/steammessages_siteserverui.proto):

```go
package main

import (
	"regexp"
	"testing"
)

var text = `
import "steammessages_base.proto";

option optimize_for = SPEED;
option cc_generic_services = false;

message SiteServerUI_Login_Request {
	optional string username = 1;
	optional string password = 2;
	optional string steamguardcode = 3;
	optional bool remember_password = 4;
}

message SiteServerUI_Login_Response {
	optional int32 logon_state = 1;
	optional int32 logon_eresult = 2;
}

message SiteServerUI_LoginStatus_Request {
}

message SiteServerUI_LoginStatus_Response {
	optional string username = 1;
	optional bool cached_credentials = 2;
	optional int32 logon_state = 3;
	optional int32 logon_eresult = 4;
}

message SiteServerUI_CancelLogin_Request {
}

message SiteServerUI_CancelLogin_Response {
	optional int32 logon_state = 1;
	optional int32 logon_eresult = 2;
}

message SiteServerUI_Logout_Request {
}

message SiteServerUI_Logout_Response {
	optional int32 logon_state = 1;
	optional int32 logout_eresult = 2;
}

message SiteServerUI_Quit_Request {
	optional bool restart = 1;
}

message SiteServerUI_Quit_Response {
}

message SiteServerUI_Status_Request {
}

message SiteServerUI_Status_Response {
	optional int32 logon_state = 1;
	optional int32 logon_eresult = 2;
	optional bool connected = 3;
	optional bool cache_enabled = 4;
	optional int32 acct_status = 5;
}

message SiteServerUI_GetLanguage_Request {
}

message SiteServerUI_GetLanguage_Response {
	optional string language = 1;
}

message SiteServerUI_SetLanguage_Request {
	optional string language = 1;
}

message SiteServerUI_SetLanguage_Response {
}

message SiteServerUI_ClientStatus_Request {
}

message SiteServerUI_ClientStatus_Response {
	message ClientInfo {
		optional uint32 ip = 1;
		optional string hostname = 2;
		optional bool connected = 3;
		optional uint64 instance_id = 4;
	}

	message Payment {
		optional uint64 transid = 1;
		optional uint64 steamid = 2;
		optional string amount = 3;
		optional int32 time_created = 4;
		optional int32 purchase_status = 5;
		optional string hostname = 6;
		optional string persona_name = 7;
		optional string profile_url = 8;
		optional string avatar_url = 9;
	}

	repeated .SiteServerUI_ClientStatus_Response.ClientInfo clients = 4;
	repeated .SiteServerUI_ClientStatus_Response.Payment payments = 5;
}

message SiteServerUI_ContentCacheStatus_Request {
}

message SiteServerUI_ContentCacheStatus_Response {
	optional bool enabled = 1;
	optional uint32 port = 2;
	optional string cache_location = 3;
	optional uint32 max_size_gb = 4;
	optional bool p2p_enabled = 5;
	optional string explicit_ip_address = 9;
	optional bool external_process = 10;
	optional uint32 current_size_gb = 6;
	optional uint64 current_bw = 7;
	optional uint64 total_bytes_served = 8;
}

message SiteServerUI_ContentCacheConfig_Request {
	optional bool enabled = 1;
	optional uint32 port = 2;
	optional string cache_location = 3;
	optional uint32 max_size_gb = 4;
	optional bool p2p_enabled = 5;
	optional bool external_process = 6;
	optional string explicit_ip_address = 7;
}

message SiteServerUI_ContentCacheConfig_Response {
}
`

// Benchmark function for the pattern \s
func BenchmarkReplaceWhitespaceSingle(b *testing.B) {
	re := regexp.MustCompile(`\s`)
	b.ResetTimer() // Start timing after setup

	for i := 0; i < b.N; i++ {
		re.ReplaceAllString(text, "")
	}
}

// Benchmark function for the pattern \s+
func BenchmarkReplaceWhitespaceMultiple(b *testing.B) {
	re := regexp.MustCompile(`\s+`)
	b.ResetTimer() // Start timing after setup

	for i := 0; i < b.N; i++ {
		re.ReplaceAllString(text, "")
	}
}
```

Results in:

```
goos: darwin
goarch: arm64
pkg: github.com/aiven/terraform-provider-aiven
BenchmarkReplaceWhitespaceSingle-10                18662             63126 ns/op
BenchmarkReplaceWhitespaceMultiple-10              18826             64126 ns/op
PASS
ok      github.com/aiven/terraform-provider-aiven       4.000s
```

`\s` took 63126 ns/op, `\s+` took 64126 ns/op.